array-destructuring/exercise-2/exercise.js

```diff
@@ -70,3 +70,22 @@ let hogwarts = [
     occupation: "Teacher",
   },
 ];
+
+let { firstName, lastName, house, pet} = hogwarts;
+
+function property(array){
+  let gryfindorHouse = array.filter(hogwarts => hogwarts.house === "Gryffindor")    //filter mean to give only the elements that pass a certain condition specified by a provided function
+  let getAllName = gryfindorHouse.map( hogwarts =>`${hogwarts.firstName}${hogwarts.lastName}`);
+  return getAllName;
+}
+console.log(property(hogwarts));
+
+function whoHavePet(array){
+  let findPet = array.filter(hogwarts => hogwarts.pet !== null)
+  let getPet = findPet.map( hogwarts =>`${hogwarts.firstName}${hogwarts.lastName}`);
+  return getPet;
+}
```

Comment:
There's some great use of `filter` and `map` functions here, Bakhat!

Just think carefully about where we could have used destructuring. For example, on line 77, we use `house` as part of the lambda function. Instead of writing `hogwarts` before `=>`, what else could you use?